### PR TITLE
Modify Chain Controller and Block Publisher to use Consensus Proxy

### PR DESCRIPTION
**Pull request by peterschwarz**
This PR combines the changes necessary to modify the validator to support consensus engines implemented with the SDK API's.

This will require the PR's for Poet and DevMode engines to properly pass tests.  Will update and rebase  as soon as those PR's are available.

### CODE REVIEW

**Comment by peterschwarz** [comment]
Based on PR #1757

**Comment by peterschwarz** [comment]
Rebased after the #1757 was merged.

File: bin/run_tests [hunk 1]
```diff
@@ -344,7 +344,7 @@ test_integration() {
     run_docker_test test_transactor_permissioning
     copy_coverage .coverage.test_transactor_permissioning
     run_docker_test test_poet_liveness
-    copy_coverage .coverage.test_poet_liveness
+    copy_coverage .coverage.test_poet_liveness --timeout 600
```
Comment (aludvik) [inline bin/run_tests]: The default is 600

File: integration/sawtooth_integration/docker/test_namespace_restriction.yaml [hunk 2]
```diff
@@ -104,6 +104,9 @@ services:
             {\\\"family\\\":\\\"xo\\\",
              \\\"version\\\":\\\"1.0\\\",
              \\\"namespaces\\\":[\\\"5b7349\\\"]}]' \
+          sawtooth.consensus.algorithm=None \
```
Comment (aludvik) [inline integration/sawtooth_integration/docker/test_namespace_restriction.yaml]: Drop the None setting

**Comment by aludvik** [comment]
See #1771

**Pull request closed without merging**